content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -278,10 +278,11 @@ manager, and scheduler run as [DaemonSet pods](/docs/concepts/workloads/controll
 configured via the Kubernetes API instead of [static pods](/docs/tasks/administer-cluster/static-pod/)
 configured in the kubelet via static files.
 
+To create a self-hosted cluster, pass the flag `--feature-gates=SelfHosting=true` to `kubeadm init`.
+
 {{< caution >}}
-**Caution:** Self-hosting is alpha, but is expected to become the default in
-a future version. To create a self-hosted cluster, pass the `--feature-gates=SelfHosting=true`
-flag to `kubeadm init`.
+**Caution:** `SelfHosting` is an alpha feature. It is deprecated in 1.12 and will be removed
+in 1.13 in favor of a different implementation that is still in the planning phase.
```

Comment:
rm "in favor of ..." -- we don't include forward-looking statements like this one.